asmcomp/cmmgen.ml

```diff
@@ -25,6 +32,21 @@ open Clambda
 open Cmm
 open Cmx_format
 
+(* Note: Closure Format
+   --------------------
+
+   The format of function closures at the cmm level is
+   {
+     code : The next curry function to call, or, at the N-1 level,
+            the actual code of the function.
+     arity : Remaining arity of the function.
+     direct : Shortcut application function. Missing at N-1 level, or
+             for unoptimized arity levels.
+     val[0] : The argument stored in this closure.
+     val[1] : Pointer to the previous closure.
```

Comment:
The comments on `val[0]` and `val[1]` make no sense.  In general we have N values that are the free variables of the closure or set of mutually recursive closures.